Tools/cases_generator/py_metadata_generator.py

```diff
@@ -0,0 +1,109 @@
+"""Generate uop metedata.
+Reads the instruction definitions from bytecodes.c.
+Writes the metadata to pycore_uop_metadata.h by default.
+"""
+
+import argparse
+import os.path
+import sys
+
+from analyzer import (
+    Analysis,
+    Instruction,
+    analyze_files,
+    Uop,
+)
+from generators_common import (
+    DEFAULT_INPUT,
+    ROOT,
+    root_relative_path,
+)
+from cwriter import CWriter
+from typing import TextIO
+
+
+def write_header(generator: str, sources: list[str], outfile: TextIO) -> None:
```

Comment:
This could be unified with the one in generators_common.py by having the comment symbol (optionally) passed in as an argument. IIRC the original code did something like that.